modules/aggregation-layers/src/utils/cpu-aggregator.js

```diff
@@ -181,12 +185,18 @@ export default class CPUAggregator {
     return result;
   }
 
-  getAggregatedData(props, viewport) {
+  getAggregatedData(props, viewport, attributes, projectPoints) {
     const aggregator = this._getAggregator(props);
 
-    // result should contain a data array and other props
-    // result = {data: [], ...other props}
-    const result = aggregator(props, viewport);
+    const {cellSize, radius, data} = props;
```

Comment:
cellSize is for grid layer, radius is for hexagon layer. explicitly call it out here is going to make cpu-aggregator less generic. the aggregator is designed to be easily override. when user pass in their own aggregator (e.g. h3 aggregation) they have access to any layer prop they need

Reply:
How is H3 aggregation uses it? With this PR, attribute data is now accessed form `attributes` array retrieved from `AttributeManager`.  Also cpu-aggregator is not documented, and my understanding was it is private to CPUGridLayer and HexagonLayer. If it is used by other layers that needs to be changed as well, this will be a breaking change.

If it helps, I can make aggregator take `props` and `opts`, where `opts` can contain additional data such as, `attributes`, `viewport` etc that are not part of `props`.

Reply:
resolved this in #3951